### PR TITLE
[MEN-4819] Disallow multiple active connections for a single device

### PR DESCRIPTION
**Pull request by alfrunes**
changelog: title

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

**Comment by alfrunes** [comment]
@mender-test-bot start pipeline

**Review by kjaskiewiczz — APPROVED** [review]
looks good to me, but it would be great if someone more familiar with deviceconnect could also review it

**Review by mchalski — APPROVED** [review]

File: api/http/device.go [hunk 1]
```diff
@@ -134,6 +134,23 @@ func (h DeviceController) Connect(c *gin.Context) {
...
+	if err != nil {
+		rest.RenderError(c,
+			http.StatusInternalServerError,
+			errors.New(http.StatusText(http.StatusInternalServerError)),
+		)
+		return
+	} else if self.Status == model.DeviceStatusConnected {
+		rest.RenderError(c,
+			http.StatusConflict, errors.Errorf(
+				"a connection already exist for device '%s'",
```
Comment (tranchitella) [inline api/http/device.go:147]:
```suggestion
				"a connection already exists for device '%s'",
```

**Review by tranchitella — APPROVED** [review]

**Comment by tranchitella** [comment]
@alfrunes no need to run again the full pipeline, imho

**Pull request closed without merging**